lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -335,6 +335,62 @@ bool SPIRVRegularizeLLVMBase::runRegularizeLLVM(Module &Module) {
   return true;
 }
 
+namespace {
+void regularizeWithOverflowInstrinsics(StringRef MangledName, CallInst *Call,
```

Comment:
I have a feeling, that this can be done easier and cheaper.
Lets imagine we have a case:
```
%res = call {i32, i1} @llvm.uadd.with.overflow.i32(i32 %a, i32 %b)
call void foo({i32, i1} %res)
```
Yes, extracting and extending i1 and storing them to a new structure will work and semantically correct. But LLVM nowadays after the switch to opaque pointers provides us strong and effective mechanisms for bit manipulations for stack allocations. And I wounder if the snippet below is semantically equivalent to the result of the work of this function:
```
%res_alloca = alloca { i32, i32 }
call spir_func void @_Z17__spirv_IAddCarryii(ptr sret(i32, i32) %res_alloca, i32 %a, i32 %b)
%res = load { i32, i1 } %res_alloca
call void foo({i32, i1} %res)
```
?

Reply:
Nice one. Yes it'll work when the second member of result can have 1 or 0 result - which is fulfilled in this case.

Reply:
I unresolve it just for other to double check the logic

Reply:
After discussion, we established that this change is unfortunately, although elegant, is non valid. The cast here it's UB in C and C++ due to type incompatibility. The source of UB here is that because value here is struct, for different types it's alignment is changeable. That means that compiler can insert hidden fields between two members of the struct and in runtime value obtained from such cast can be undefined.